process/dataAggregator.go

```diff
@@ -0,0 +1,62 @@
+package process
+
+import (
+	"encoding/hex"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+	"github.com/multiversx/mx-chain-ws-connector-template-go/data"
+)
+
+type dataAggregator struct {
+	blocksPool BlocksPool
+}
+
+// NewDataAggregator will create a new data aggregator instance
+func NewDataAggregator(
+	blocksPool BlocksPool,
+) (*dataAggregator, error) {
+	if check.IfNil(blocksPool) {
+		return nil, ErrNilBlocksPool
+	}
+
+	return &dataAggregator{
+		blocksPool: blocksPool,
+	}, nil
+}
+
+// ProcessHyperBlock will process meta outport block. It will try to fetch and aggregate
+// notarized shards data
+func (da *dataAggregator) ProcessHyperBlock(outportBlock *outport.OutportBlock) (*data.HyperOutportBlock, error) {
+	hyperOutportBlock := &data.HyperOutportBlock{}
+	hyperOutportBlock.MetaOutportBlock = outportBlock
```

Comment:
maybe some check that the received outport block is indeed a metablock, e.g. check the ShardID?

Reply:
added shard id check